lib/SILOptimizer/Mandatory/MoveOnlyAddressCheckerUtils.cpp

```diff
@@ -2971,15 +2974,15 @@ void ExtendUnconsumedLiveness::runOnField(
     }
 
     // Walk backwards from consuming blocks.
+    BasicBlockBitfield cache(currentDef->getFunction(), 2);
     while (auto *block = worklist.pop()) {
-      if (!originalLiveBlocks.insert(block)) {
+      if (!originalLiveBlocks.insert(block)
+          // Don't walk into the predecessors of blocks which kill liveness.
+          && !isLiveAtBegin(block, element, /*isLiveAtEnd=*/true, destroys,
```

Comment:
As discussed, we probably don't need to check local liveness for each block on the worklist. So you probably then don't even need the cache,

Reply:
Fixed.